src/serac/numerics/functional/functional.hpp

```diff
@@ -375,12 +396,43 @@ class Functional<test(trials...), exec> {
     static_assert(sizeof...(T) == num_trial_spaces,
                   "Error: Functional::operator() must take exactly as many arguments as trial spaces");
 
-    [[maybe_unused]] constexpr int                                           wrt = index_of_differentiation<T...>();
-    std::array<std::reference_wrapper<const mfem::Vector>, num_trial_spaces> input_T{args...};
+    [[maybe_unused]] constexpr int                          wrt = index_of_differentiation<T...>();
+    std::vector<std::reference_wrapper<const mfem::Vector>> input_T{args...};
+
+    return (*this)(input_T, Index<wrt>{});
+  }
 
+  /**
+   * @brief this function lets the user evaluate the serac::Functional with the given trial space values
+   *
+   * note: it accepts a vector of mfem::Vectors that must be of length `num_trial_spaces`. This interface
+   * assumes no derivative information is needed.
+   *
+   * @param input_T an array of trial space dofs used to carry out the calculation.
+   */
+  mfem::Vector& operator()(std::vector<std::reference_wrapper<const mfem::Vector>> input_T)
```

Comment:
Can we use `std::array` instead of `std::vector` here? The number of entries is known at compile time.

Reply:
I tried to make an array at some point and got tangled in an issue of default values in a constructor. I've cleaned things up a bit since then, so I'll give it another shot.